src/bosh.js

```diff
@@ -632,6 +636,9 @@ Strophe.Bosh.prototype = {
                      this._requests[0].age() > Math.floor(Strophe.SECONDARY_TIMEOUT * this.wait))) {
                     this._restartRequest(0);
                 }
+
+                this._conn.ridChange(Number(req.rid) + 1);
+
```

Comment:
When I was working on the session restore feature in #119, I also initially stored the RID every time it got incremented, instead of storing it only when the page unloads.

I called `_cacheSession` in `_buildBody`, and it worked well. See here: https://github.com/strophe/strophejs/blob/d2e7c270e35dc73c6f2013eb3c5e5441015a6572/src/bosh.js#L197

I don't really understand what you mean by this:
>  It is mandatory to save the rid after a successful request, because the server does the same.

All that's required is that you cache either:
1. the last used RID
2. an unused RID value that is at least 1 higher than the last used RID

> Maybe you feel better if we rename the function to something like nextValidRid.

Well yes, the name does better imply what's going on.

I'm curious why you don't just use the `restore` feature I already added in #119. It's already deployed in production and works well and I've been using the same approach with converse.js for about a year now, without any complaints.